src/app/dynamic-forms/cinchy-dynamic-forms.component.ts

```diff
@@ -583,52 +593,49 @@ export class CinchyDynamicFormsComponent implements OnInit, OnChanges, OnDestroy
 
     return new Promise((resolve, reject) => {
 
-      if (this.childDataForm.length === 0 && !isNullOrUndefined(this.rowId)) {
+      if (!this.childDataForm.length && !isNullOrUndefined(this.rowId)) {
         this.getSavedData();
-        resolve();
 
-        return;
+        resolve();
       }
-
-      if (this.childDataForm.length <= idx) {
+      else if (this.childDataForm.length <= idx) {
         resolve();
-
-        return;
```

Comment:
Would removing the resolve cause issues in the future? Despite it looking cleaner, it's a safeguard to end the logic early in-case someone in the future decides to add more to the end of the function, the logic will continue after the resolve.